pkg/controllers/infracluster/infracluster_controller.go

```diff
@@ -214,6 +217,13 @@ func (r *InfraClusterController) ensureInfraCluster(ctx context.Context, log log
 		if err != nil {
 			return nil, fmt.Errorf("error getting InfraCluster object: %w", err)
 		}
+	case configv1.BareMetalPlatformType:
+		baremetalCluster := &metal3v1.Metal3Cluster{}
+		if err := r.Get(ctx, client.ObjectKey{Namespace: defaultCAPINamespace, Name: r.Infra.Status.InfrastructureName}, baremetalCluster); err != nil && !kerrors.IsNotFound(err) {
+			return nil, fmt.Errorf("error getting InfraCluster object: %w", err)
+		}
+
+		infraCluster = baremetalCluster
```

Comment:
This will need changing once we introduce the BareMetalCluster generation via this controller. In a followup PR if possible @honza, thank you!